flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/output/SnapshotUtilsTest.java

```diff
@@ -60,6 +62,25 @@ public void testSnapshotUtilsLifecycle() throws Exception {
         Assert.assertEquals(EXPECTED_CALL_OPERATOR_SNAPSHOT, ACTUAL_ORDER_TRACKING);
     }
 
+    @Test
+    public void testSnapshotUtilsLifecycleWithNativeSavepoint() throws Exception {
```

Comment:
I don't think this test actually test the snapshot type. I think we can recored the `snapshotType` in `LifecycleOperator#snapshotState` method, and then verify that we have passed the native savepoint snapshot type.